Adds support for trogon.

boehmseb opened the pull request:
Most vara-* commands now support `tui` as a subcommand which opens a text-based UI for selecting command line parameters.

Some limitations apply:
- `vara-run` is not supported because arguments do not play nice with command groups in click
- `vara-plot` and `vara-table` require support for `click.MultiCommand` (Textualize/trogon#71)
- `trogon` contains some more small bugs that limit usability. I opened PRs for them, but until they are merged, you can use the following branch for testing: https://github.com/boehmseb/trogon/tree/trogonFixes

boehmseb committed:
Adds support for trogon.

Most vara-* commands now support `tui` as a subcommand which opens a text-based UI for selecting command line parameters.

vulder committed:
Merge branch 'vara-dev' into f-Trogon

boehmseb committed:
vara-art: make artefacts a click.Choice

boehmseb committed:
mypy: Ignore missing imports for trogon

boehmseb committed:
vara-cs: make project sleection a click.Choice

LuAbelt commented on varats/varats/tools/driver_container.py:
```diff
@@ -144,7 +146,8 @@ def delete(
     "--tool",
     type=click.Choice([*get_supported_research_tool_names(), "none"]),
-    default=lambda: vara_cfg()["container"]["research_tool"].value or "none",
+    required=True,
+    # default=lambda: vara_cfg()["container"]["research_tool"].value or "none",
```
What is this change for?

boehmseb replied:
Trogon currently cannot evaluate functions as default parameters. Since we can determine a good default only dynamically, I made the option required instead and removed the default.

boehmseb committed:
vara-pc: make paper configs a click.Choice and simplify paper config handling.

Many of the checks are not necessary anymore because our config management takes care of managing folders much better.

boehmseb committed:
suppress mypy errors

boehmseb committed:
suppress mypy errors

boehmseb committed:
Revert trogon support for vara-run

boehmseb committed:
Fix vara-gen-bbconfig tui support

boehmseb committed:
vara-pc select: paper config is optional

boehmseb committed:
Fix vara-art show

vulder commented on varats/varats/tools/driver_paper_config.py:
```diff
@@ -32,85 +34,56 @@ def main() -> None:
+def _pc_set(paper_config: tp.Optional[str]) -> None:
     if not paper_config:
-        pc_folder_path = Path(
-            get_value_or_default(
-                vara_cfg()["paper_config"], "folder",
-                str(get_varats_base_folder())
-            )
-        )
-        if not (pc_folder_path.exists() and pc_folder_path.is_dir()):
-            LOG.error(
-                f"Paper config folder not set: {pc_folder_path} "
-                "(Path does not exist or is no directory)."
-            )
-            return
```
Do we no longer need the checks here?

boehmseb replied:
No, because by now we take care of creating these directories whenever we save the config.

boehmseb replied:
And the config folder has a sane default value set.

vulder reviewed (commented):
Do I understand this correctly, this is "opt-in" as in one can use the CLI as is, without changes and if wanted with the clicky UI?

PS: also some tests seem not to be happy :D

boehmseb commented:
> Do I understand this correctly, this is "opt-in" as in one can use the CLI as is, without changes and if wanted with the clicky UI?
Yes, this basically adds a new subcommand `tui` to each supported tool that starts the text-based UI (bonus points for a good idea where/how to document this).
 
> PS: also some tests seem not to be happy :D
I have no idea why. That test initially also failed on my machine but for a different reason. All tests pass when I execute them locally.

boehmseb committed:
Fix artefact driver test.

The reason for the failing test was that the CLI parameters are generated at parse-time. Since the test modified the available choices of such an option, we need to reload the driver module for that change to take effect at the CLI level.
